data/mp/multiplay/skirmish/bonecrusher/functions.js

```diff
@@ -0,0 +1,1215 @@
+// 	for (var x in _globalInfoNear){
+// 	debugMsg("x = "+Object.keys(_globalInfoNear), 'stats');
+// 		level2+=_globalInfoNear[x].length;
+// 		debugMsg("y = "+Object.keys(_globalInfoNear[x]), 'stats');
+// 		for (var y in _globalInfoNear[x]){
+// 			level3+=1;
+// 			debugMsg("z = "+Object.keys(_globalInfoNear[x][y]), 'stats');
+// 			for (var z in _globalInfoNear[x][y]){
+// 				level4+=3;
+// 				debugMsg("keys = "+Object.keys(_globalInfoNear[x][y][z]), 'stats');
+// 				
+// 			}
+// 		}
+// 	}
+// 	debugMsg("_globalInfoNear level1 = "+level1, 'stats');
+// 	debugMsg("_globalInfoNear level2 = "+level2, 'stats');
+// 	debugMsg("_globalInfoNear level3 = "+level3, 'stats');
+// 	debugMsg("_globalInfoNear level4 = "+level4, 'stats');
+	
+//	enumFeature(ALL_PLAYERS, "").filter(function(e){if(e.armour == 5 && e.thermal == 5 && !e.damageable && e.health == 100 && e.player==12)return true;return false;})
+//	.forEach(function(e){
+//		debugMsg(e.name+' '+e.x+'x'+e.y+'; type='+e.type+'; id='+e.id+'; player='+e.player+'; selected='+e.selected+'; health='+e.health+'; armour='+e.armour+'; thermal='+e.thermal+'; damageable='+e.damageable, 'stats');
+//	});
+}
+
+//Функция определяет подвергается ли ремонту наша цель
+//Если да, возвращяем объект, кто ремонтирует нашу цель
+function isBeingRepaired(who){
+	if ( typeof who === "undefined" ) {
+//		debugMsg("Атакующий неизвестен",5);
+		return false;
+	}
+
+	if ( allianceExistsBetween(me, who.player) ) {
+//		debugMsg("Атакующий мой союзник, прощаю",5);
+		return false;
+	}
+
+	switch ( who.type ) {
+		case DROID: {
+//			debugMsg("Нас атаковал вражеский дроид ["+who.player+"]",5);
+//TODO
+//			Тут нужно доработать
+			return false;
+
+		}
+		case STRUCTURE: {
+//			debugMsg("Нас атакует вражесая башня ["+who.player+"]",5);
+			var droids = enumDroid(who.player,DROID_CONSTRUCT,me);
+			if ( droids.length != 0 ) {
+				for ( var i in droids ) {
+					if ( distBetweenTwoPoints_p(who.x,who.y,droids[i].x,droids[i].y) <= 3 ) {
+//						debugMsg("Атакующая меня башня подвергается ремонту!",5);
+						return droids[i];
+					}
+				}
+			}	
+			return false;
+
+		}
+		default: {
+//			debugMsg("Нууу, не знаю, нас атакует Ктулху! ["+who.player+"]",5);
+			return false;
+
+		}
+	}
+
+}
+
+//Какой-то бардак в функциях движка
+//enumResearch() - возвращает список ТОЛЬКО тех, которые можем стартануть
+//getResearch() - возвращет объект любой запрошенной технологии
+//объект содержит .done и .started которые отвечают за завершённую или запущенную технологию
+//почему бы не сделать что бы enumResearch() возвращал объект ВСЕХ технологий
+//в котором бы содержалась значения .done .started и конежно же .available
+//вместо того что сделано сейчас, можно было бы легко использовать filter()
+//В голову не идёт как это исправить своими силами, забил пока..
+function _doResearch(){
+	debugMsg("doResearch()", 'research');
+	var avail_research = enumResearch().filter(function(e){if(e.started)return false;return true;});
```

Comment:
This is a global used in other parts of the scripts. Remove `var`.

Reply:
This non-used function, i better to purged function.